core/src/main/java/org/infinispan/interceptors/impl/BaseAsyncInvocationContext.java

```diff
@@ -0,0 +1,580 @@
+package org.infinispan.interceptors.impl;
+
+import org.infinispan.commands.VisitableCommand;
+import org.infinispan.commons.CacheException;
+import org.infinispan.context.AsyncInvocationContext;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.interceptors.AsyncInterceptor;
+import org.infinispan.util.concurrent.CompletableFutures;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+import org.jboss.logging.NDC;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * This base class implements the {@link AsyncInvocationContext} methods.
+ *
+ * @author Dan Berindei
+ * @since 9.0
+ */
+public abstract class BaseAsyncInvocationContext implements InvocationContext {
+   private static final Log log = LogFactory.getLog(BaseAsyncInvocationContext.class);
+   private static final boolean trace = log.isTraceEnabled();
+   // Enable when debugging BaseAsyncInvocationContext itself
+   private static final boolean EXTRA_LOGS = SecurityActions.getBooleanProperty("org.infinispan.debug.BaseAsyncInvocationContext");
+
+   private static final CompletableFuture<Void> CONTINUE_INVOCATION = CompletableFuture.completedFuture(null);
```

Comment:
It's already cached :)
The reason I wanted to use a different instance here is because I may want to add a special case for interceptors that return `BaseAsyncInvocationContext.continueInvocation()` in the future.